programs/sbf/rust/account_mem/src/lib.rs

```diff
@@ -108,7 +108,20 @@ pub fn process_instruction(
             // memmov dst overlaps begin of account
             unsafe { sol_memmove(too_early(3).as_mut_ptr(), buf.as_ptr(), 10) };
         }
-
+        14 => {
+            // memmove dst overlaps begin of account, reverse order
+            unsafe { sol_memmove(too_early(0).as_mut_ptr(), too_early(3).as_ptr(), 10) };
+        }
+        15 => {
+            // memmove dst overlaps end of account, reverse order
+            unsafe {
+                sol_memmove(
```

Comment:
which pointer here is causing the failure? I think they are both wrong? what
about the case where dst is ok but src is wrong? and vice versa?

Reply:
the destination falls within src + length

Reply:
data[data_len..].as_mut_ptr() + 10 <- isn't this outside the account region?

Reply:
Sorry, I misread your comment. I've added a bunch of tests: https://github.com/anza-xyz/agave/pull/4650